### PR TITLE
Feat #8 

### LINKED ISSUE #8: Expand / Collapse functionality
Introduce expand / collapse functionality like the sidebar that we have in every project.

This can be a component or a helper.

### PR DESCRIPTION
**Pull request by mkarajohn**
Please provide feedback for the initial implementation

Closes #8

### CODE REVIEW

**Comment by mkarajohn** [comment]
@panvourtsis I defined the `TestId` type in the `global.d.ts` file, let me know if we want it somewhere else

File: src/components/ExpandCollapse/ExpandCollapse.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    } else {
+      timeout = window.setTimeout(() => {
+        if (contentRef.current === null) {
+          throw new Error('Uninitialised element ref');
+        }
+        contentRef.current.style.visibility = 'hidden';
+      }, transitionDuration);
+    }
+
+    return function() {
```
Comment (panvourtsis) [inline src/components/ExpandCollapse/ExpandCollapse.tsx:139]: also seeing a named function with this kind of return confused me for some time

File: src/components/ExpandCollapse/ExpandCollapse.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      contentRef.current.style.height = `0`;
+    }
+  }
+
+  function manageContentVisibility() {
+    if (contentRef.current === null) {
+      throw new Error('Uninitialised element ref');
+    }
+
+    let timeout: number;
```
[the remark this answers is not in the document]
Comment (mkarajohn) [reply src/components/ExpandCollapse/ExpandCollapse.tsx:126]: Yes, `setTimeout` returns a number ID

File: src/components/ExpandCollapse/ExpandCollapse.tsx (hunk 1 above)
Comment (mkarajohn) [reply src/components/ExpandCollapse/ExpandCollapse.tsx:139]: Non-issue

**Comment by mkarajohn** [comment]
Actually I will rework this in order to implement some things I've learnt during this time

**Comment by panvourtsis** [comment]
@mkarajohn is this still active?

**Comment by mkarajohn** [comment]
Yes, in the backburner for now. I have some improvements to make

**Comment by panvourtsis** [comment]
can we plan it to close it as it will be forgotten

**Comment by mkarajohn** [comment]
Frick this, resolving remaining comments and will revisit in the future

**Review by panvourtsis — APPROVED** [review]

**Pull request merged**